src/adapters/api-blueprint-adapter.coffee

```diff
@@ -298,9 +297,9 @@ legacyResourcesFrom1AResource = (legacyUrlConverterFn, resource, sourcemap, opti
       legacyResource.model = {}
 
     legacyResource.resourceParameters = resourceParameters or []
-    legacyResource.actionParameters   = actionParameters or []
+    legacyResource.actionParameters   = getParametersOf(action, options) or []
 
-    legacyResource.parameters = actionParameters or resourceParameters or []
+    legacyResource.parameters = legacyResource.resourceParameters.concat(legacyResource.actionParameters)
```

Comment:
I think it should be the following:

```coffeescript
# explicit parameters from our parent resource
legacyResource.resourceParameters = resourceParameters or []

# the parameters explicitly set on the parameter
legacyResource.actionParameters   = actionParameters or []

# convenience to get the parameters to use for this resource or action
legacyResource.parameters = actionParameters or resourceParameters
```

**NOTE:** *What a `legacyResource` is in Apiary Blueprint and Application AST isn't what we call a resource, but actually an action.*